core/src/main/kotlin/org/opensearch/alerting/core/lock/LockService.kt

```diff
@@ -72,13 +75,23 @@ class LockService(private val client: Client, private val clusterService: Cluste
                                 object : ActionListener<LockModel> {
                                     override fun onResponse(existingLock: LockModel?) {
                                         if (existingLock != null) {
+                                            val currentTimestamp = getNow()
                                             if (isLockReleased(existingLock)) {
                                                 log.debug("lock is released or expired: {}", existingLock)
                                                 val updateLock = LockModel(existingLock, getNow(), false)
                                                 updateLock(updateLock, listener)
                                             } else {
-                                                log.debug("Lock is NOT released or expired. {}", existingLock)
-                                                listener.onResponse(null)
+                                                log.debug("Lock is NOT released. {}", existingLock)
```

Comment:
plz add a test to verify expired locks are retrieved and deleted/re-used instead. create in now()- 10 mins

Reply:
added a test. https://github.com/opensearch-project/alerting/pull/1623/files#diff-9b08d53e8cff3d739beb6ba304e4eaf466f08412762c3cd55886a52b722a77f1R2753